src/Stratis.FederatedPeg.Features.FederationGateway/Notifications/BlockObserver.cs

```diff
@@ -64,11 +71,27 @@ protected override void OnNextCore(ChainedHeaderBlock chainedHeaderBlock)
         {
             this.walletSyncManager.ProcessBlock(chainedHeaderBlock.Block);
 
-            this.federationGatewayClient.PushCurrentBlockTipAsync(
-                new BlockTipModel(
-                    chainedHeaderBlock.ChainedHeader.HashBlock,
-                    chainedHeaderBlock.ChainedHeader.Height,
-                    (int)this.depositExtractor.MinimumDepositConfirmations)).ConfigureAwait(false).GetAwaiter().GetResult();
+            // Cancel previous sending to avoid first sending new tip and then sending older tip.
+            if ((this.pushBlockTipTask != null) && !this.pushBlockTipTask.IsCompleted)
+            {
+                this.cancellationSource.Cancel();
+                this.pushBlockTipTask.GetAwaiter().GetResult();

```

Comment:
```suggestion
                this.pushBlockTipTask.ConfigureAwait(false).GetAwaiter().GetResult();
```

Reply:
`ConfigureAwait(false)` let's task know that it can return to another thread after the execution. 
`.GetAwaiter().GetResult();` makes the task run in blocking mode in this very thread. 

So there is no reason to use `ConfigureAwait(false)` when we are doing `.GetAwaiter().GetResult();`

Reply:
If my understanding of multi-threading using tasks is wrong please provide an article that will prove that calling `.ConfigureAwait(false).GetAwaiter().GetResult();` makes sense.
I'm not 100% expert on this but I think it works as I wrote above.